peer.go

```diff
@@ -1602,7 +1612,7 @@ out:
 
 			// Create the link and add it to the switch.
 			err = p.addLink(
-				chanPoint, newChan, forwardingPolicy,
+				chanPoint, lnChan, forwardingPolicy,
```

Comment:
Similar to the case above, we're missing a call to `lnChan.Stop()` here in the case that we can't actually add the link to the switch.

Reply:
That's interesting, it sounds correct. Who were responsible for stopping the channel before this change, as it seems to be the exact same problem already?

It also looks like any error here is never communicated back the the caller, so that should probably be fixed as well.